src/ripple/app/paths/impl/DirectStep.cpp

```diff
@@ -691,7 +710,7 @@ make_DirectStepI (
 {
     // Only charge a transfer fee if the previous step redeems
     auto r = std::make_unique<DirectStepI> (
-        src, dst, c, ctx.prevStep, ctx.j);
+        src, dst, c, ctx.prevStep, ctx.isLast, ctx.j);
```

Comment:
I've been meaning to ask a general question about the `*Step` constructors.  It seems like the `make_*Step` methods pull out bits of the passed in `StrandContext` and use them to call the `*Step` constructors.

Not for this pull request, but what would you think about passing the entire `StrandContext` to the `*Step` constructors and letting each constructor choose the pieces it wants for itself.  The advantage would be fewer arguments passed to the constructors.  Just for your future consideration.  I could potentially make this change as part of my offer crossing work.